scripts/generate_18s_its_primer_sql

```diff
@@ -0,0 +1,1644 @@
+GAGTTAGCATCA	CGGCTGCGTTCTTCATCGATGC	8	C12
+TGTAGTATAGGC	CGGCTGCGTTCTTCATCGATGC	8	D1
+CTCACGCAATGC	CGGCTGCGTTCTTCATCGATGC	8	D2
+GTCCCGTGAAAT	CGGCTGCGTTCTTCATCGATGC	8	D3
+GGACAGTGTATT	CGGCTGCGTTCTTCATCGATGC	8	D4
+ACACGACTATAG	CGGCTGCGTTCTTCATCGATGC	8	D5
+GTGTAGGTGCTT	CGGCTGCGTTCTTCATCGATGC	8	D6
+TGAACTAGCGTC	CGGCTGCGTTCTTCATCGATGC	8	D7
+TCCGAGTCACCA	CGGCTGCGTTCTTCATCGATGC	8	D8
+TCCTCTTTGGTC	CGGCTGCGTTCTTCATCGATGC	8	D9
+TCCACCCTCTAT	CGGCTGCGTTCTTCATCGATGC	8	D10
+TCGTGACGCTAA	CGGCTGCGTTCTTCATCGATGC	8	D11
+ACGGCTAGTTCC	CGGCTGCGTTCTTCATCGATGC	8	D12
+GCACTGGCATAT	CGGCTGCGTTCTTCATCGATGC	8	E1
+GGCATTAGTTGA	CGGCTGCGTTCTTCATCGATGC	8	E2
+CGGTAGTTGATC	CGGCTGCGTTCTTCATCGATGC	8	E3
+TGAAAGCGGCGA	CGGCTGCGTTCTTCATCGATGC	8	E4
+GGTTACGGTTAC	CGGCTGCGTTCTTCATCGATGC	8	E5
+ACATCAGGTCAC	CGGCTGCGTTCTTCATCGATGC	8	E6
+GTTGATACGATG	CGGCTGCGTTCTTCATCGATGC	8	E7
+CAGACACTTCCG	CGGCTGCGTTCTTCATCGATGC	8	E8
+TCACCATCCGAG	CGGCTGCGTTCTTCATCGATGC	8	E9
+ACCCACCACTAG	CGGCTGCGTTCTTCATCGATGC	8	E10
+CAGAAGGTGTGG	CGGCTGCGTTCTTCATCGATGC	8	E11
+GAAGCTTGAATC	CGGCTGCGTTCTTCATCGATGC	8	E12
+ACTAGGATCAGT	CGGCTGCGTTCTTCATCGATGC	8	F1
+GCTCCTTAGAAG	CGGCTGCGTTCTTCATCGATGC	8	F2
+TCCCATTCCCAT	CGGCTGCGTTCTTCATCGATGC	8	F3
+TGGCGTCATTCG	CGGCTGCGTTCTTCATCGATGC	8	F4
+AATCCTCGGAGT	CGGCTGCGTTCTTCATCGATGC	8	F5
+CTGGACGCATTA	CGGCTGCGTTCTTCATCGATGC	8	F6
+ACCGATTAGGTA	CGGCTGCGTTCTTCATCGATGC	8	F7
+ATGTGCTGCTCG	CGGCTGCGTTCTTCATCGATGC	8	F8
+TACGTACGAAAC	CGGCTGCGTTCTTCATCGATGC	8	F9
+ATCACATTCTCC	CGGCTGCGTTCTTCATCGATGC	8	F10
+AGCCTGGTACCT	CGGCTGCGTTCTTCATCGATGC	8	F11
+GCTAAAGTCGTA	CGGCTGCGTTCTTCATCGATGC	8	F12
+TCTCAGCGCGTA	CGGCTGCGTTCTTCATCGATGC	8	G1
+GACCCTAGACCT	CGGCTGCGTTCTTCATCGATGC	8	G2
+TATTCAGCGGAC	CGGCTGCGTTCTTCATCGATGC	8	G3
+GTTCCGGATTAG	CGGCTGCGTTCTTCATCGATGC	8	G4
+GCGTGTAATTAG	CGGCTGCGTTCTTCATCGATGC	8	G5
+CTGTAGCTTGGC	CGGCTGCGTTCTTCATCGATGC	8	G6
+ATGCCTCGTAAG	CGGCTGCGTTCTTCATCGATGC	8	G7
+ACCTATGGTGAA	CGGCTGCGTTCTTCATCGATGC	8	G8
+CTGTTACAGCGA	CGGCTGCGTTCTTCATCGATGC	8	G9
+CAGTCAGGCCTT	CGGCTGCGTTCTTCATCGATGC	8	G10
+ACTGAGCTGCAT	CGGCTGCGTTCTTCATCGATGC	8	G11
+ACGAAGTCTACC	CGGCTGCGTTCTTCATCGATGC	8	G12
+ACCGTCTTTCTC	CGGCTGCGTTCTTCATCGATGC	8	H1
+AGTCTGTCTGCG	CGGCTGCGTTCTTCATCGATGC	8	H2
+CCGCACTCAAGT	CGGCTGCGTTCTTCATCGATGC	8	H3
+TGTGGAAACTCC	CGGCTGCGTTCTTCATCGATGC	8	H4
+TTAGGCAGGTTC	CGGCTGCGTTCTTCATCGATGC	8	H5
+TAAGACTACTGG	CGGCTGCGTTCTTCATCGATGC	8	H6
+CGCGAAGTTTCA	CGGCTGCGTTCTTCATCGATGC	8	H7
+CGATACACTGCC	CGGCTGCGTTCTTCATCGATGC	8	H8
+TTGAAATCCCGG	CGGCTGCGTTCTTCATCGATGC	8	H9
+GTTAGGGAGCGA	CGGCTGCGTTCTTCATCGATGC	8	H10
+TTACTGTGGCCG	CGGCTGCGTTCTTCATCGATGC	8	H11
+ATATAAGGCCCA	CGGCTGCGTTCTTCATCGATGC	8	H12￿￿￿￿￿
+"""
+
+
+def generate_insert_values(last_plate_id, last_container_id):
+    set_definitions_by_name = {"EMP 18S": emp_18s_tsvs,
+                               "EMP ITS": emp_its_tsvs}
+    # from records created by db_patch_manual.sql
+    primer_set_ids_by_name = {"EMP 18S": 3, "EMP ITS": 4}
+
+    num_plates_in_set = 8
+    num_rows_on_plate = 8
+    num_cols_on_plate = 12
+    num_wells_on_plate = num_ro
```

Comment:
I don't understand the hardcoded values here, would these ever cause a real problem in practice? It seems like this is a use once and burn it script, right?

Reply:
Hmm, I don't think they'd cause a problem but I'll refactored the script so they are default arguments to the function rather than hard-coded constants inside the function.

This IS basicaly a "use once and burn it" script, but I checked it into the repo (rather than just trashing it) because I figured that if we ever came up with ANOTHER primer set (and therefore needed to input it into the db) it would be easier to modify this script than to rewrite the same functionality again--which I think we would definitely end up doing, because writing the SQL to fill out all these tables by hand for multiple plates would be *at least* as error-prone as scripting it out.